From 0fee1cf178f455cbeac5ff734effba24f5326eec Mon Sep 17 00:00:00 2001
From: Wang Haoran <b50027@freescale.com>
Date: Wed, 29 Oct 2014 09:04:17 +0800
Subject: [PATCH 365/602] MA-5791 Force Atheros wifi use channel 1 in p2p

As we know AOSP wpa_supplicant_8 of Kitkat has issue
with p2p in 5G. In 2.4G it choose random p2p channel
if not assign it. So choose channel 1 to be the p2p
operate channel that will improve the compatibility.

Signed-off-by: Wang Haoran <b50027@freescale.com>
---
 common/wifi/p2p_supplicant_advance_overlay.conf | 2 --
 evk_6sl/init.rc                                 | 2 +-
 sabreauto_6q/init.rc                            | 2 +-
 sabreauto_6sx/init.rc                           | 2 +-
 sabresd_6dq/init.rc                             | 2 +-
 sabresd_6sx/init.rc                             | 2 +-
 6 files changed, 5 insertions(+), 7 deletions(-)

diff --git a/common/wifi/p2p_supplicant_advance_overlay.conf b/common/wifi/p2p_supplicant_advance_overlay.conf
index 565b73c..b4d4f5e 100644
--- a/common/wifi/p2p_supplicant_advance_overlay.conf
+++ b/common/wifi/p2p_supplicant_advance_overlay.conf
@@ -1,5 +1,3 @@
 disable_scan_offload=1
-p2p_listen_reg_class=81
-p2p_listen_channel=1
 p2p_oper_reg_class=81
 p2p_oper_channel=1
diff --git a/evk_6sl/init.rc b/evk_6sl/init.rc
index c4ffa5f..cc38df7 100644
--- a/evk_6sl/init.rc
+++ b/evk_6sl/init.rc
@@ -104,7 +104,7 @@ on boot
 
 service p2p_supplicant /system/bin/wpa_supplicant \
     -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf \
-    -I/system/etc/wifi/p2p_supplicant_overlay.conf -N \
+    -I/system/etc/wifi/p2p_supplicant_advance_overlay.conf -N \
     -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
     -I/system/etc/wifi/wpa_supplicant_overlay.conf \
     -O/data/misc/wifi/sockets -puse_p2p_group_interface=1 \
diff --git a/sabreauto_6q/init.rc b/sabreauto_6q/init.rc
index 63b94c7..45e8bbc 100644
--- a/sabreauto_6q/init.rc
+++ b/sabreauto_6q/init.rc
@@ -107,7 +107,7 @@ on boot
 
 service p2p_supplicant /system/bin/wpa_supplicant \
     -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf \
-    -I/system/etc/wifi/p2p_supplicant_overlay.conf -N \
+    -I/system/etc/wifi/p2p_supplicant_advance_overlay.conf -N \
     -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
     -I/system/etc/wifi/wpa_supplicant_overlay.conf \
     -O/data/misc/wifi/sockets -puse_p2p_group_interface=1 \
diff --git a/sabreauto_6sx/init.rc b/sabreauto_6sx/init.rc
index ef304aa..003c1df 100755
--- a/sabreauto_6sx/init.rc
+++ b/sabreauto_6sx/init.rc
@@ -105,7 +105,7 @@ on boot
 
 service p2p_supplicant /system/bin/wpa_supplicant \
     -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf \
-    -I/system/etc/wifi/p2p_supplicant_overlay.conf -N \
+    -I/system/etc/wifi/p2p_supplicant_advance_overlay.conf -N \
     -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
     -I/system/etc/wifi/wpa_supplicant_overlay.conf \
     -O/data/misc/wifi/sockets -puse_p2p_group_interface=1 \
diff --git a/sabresd_6dq/init.rc b/sabresd_6dq/init.rc
index e2649b3..478380b 100755
--- a/sabresd_6dq/init.rc
+++ b/sabresd_6dq/init.rc
@@ -135,7 +135,7 @@ on charger
 
 service p2p_supplicant /system/bin/wpa_supplicant \
     -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf \
-    -I/system/etc/wifi/p2p_supplicant_overlay.conf -N \
+    -I/system/etc/wifi/p2p_supplicant_advance_overlay.conf -N \
     -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
     -I/system/etc/wifi/wpa_supplicant_overlay.conf \
     -O/data/misc/wifi/sockets -puse_p2p_group_interface=1 \
diff --git a/sabresd_6sx/init.rc b/sabresd_6sx/init.rc
index afffb3f..bbeb874 100755
--- a/sabresd_6sx/init.rc
+++ b/sabresd_6sx/init.rc
@@ -120,7 +120,7 @@ on charger
 
 service p2p_supplicant /system/bin/wpa_supplicant \
     -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf \
-    -I/system/etc/wifi/p2p_supplicant_overlay.conf -N \
+    -I/system/etc/wifi/p2p_supplicant_advance_overlay.conf -N \
     -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
     -I/system/etc/wifi/wpa_supplicant_overlay.conf \
     -O/data/misc/wifi/sockets -puse_p2p_group_interface=1 \
-- 
1.8.0

